types/event.go

```diff
@@ -33,6 +33,53 @@ type Event struct {
 	Data        map[string]interface{}
 }
 
+// GetDataSize computes the size of the Data element of the Event.
+func (e *Event) GetDataSize() int {
+	total := 0
+	for _, v := range e.Data {
+		total += getByteSize(v)
+	}
+	return total
+}
+
+// getByteSize returns the size of the given value in bytes.
+// This is a rough estimate, but it's good enough for our purposes.
+// Maps and slices are the most complex, so we'll just add up the sizes of their entries.
+func getByteSize(val any) int {
+	switch value := val.(type) {
+	case bool:
+		return 1
+	case float64, int64, int:
+		return 8
+	case string:
+		return len(value)
+	case []byte: // also catch []uint8
+		return len(value)
+	case []bool:
+		return len(value)
+	case []float64:
```

Comment:
I don't think it hurts to look at all these slice types, but I don't think that in practice you're ever going to get anything other than a `[]any` which you don't handle here explicitly. You definitely need that.

Reply:
I say that because the JSON unmarshaler won't do anything else and I'm pretty sure that Husky converts all array types into []any as well.

Reply:
I did originally have `[]any` in the list, but in tests I never hit that case. Maybe I need to add a test that explicitly uses `[]any` as the data entry type.

Reply:
Added case for `[]any` in [236113e](https://github.com/honeycombio/refinery/pull/1510/commits/236113e1708c4855c374fe523ac6f57270ca13b3)

Reply:
Even if it doesn't go into the automated tests, it might be worth doing a manual test that sends, say, a []int in a field through the OTLP endpoint and see how it shows up in this function.

Reply:
You're right about [Husky](https://github.com/honeycombio/husky/blob/main/otlp/common.go#L408) using []interface as for array types. I'll confirm with the JSON unmarshaller and may remove the other cases.